test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatLocalsTests.cs

```diff
@@ -66,11 +100,19 @@ public static void Locals_Succeeds(string args)
                     {
                         { "NUGET_PACKAGES", mockGlobalPackagesDirectory.FullName },
                         { "NUGET_HTTP_CACHE_PATH", mockHttpCacheDirectory.FullName },
-                        { RuntimeEnvironmentHelper.IsWindows?"TMP":"TMPDIR", mockTmpCacheDirectory.FullName }
+                        { RuntimeEnvironmentHelper.IsWindows ? "TMP" : "TMPDIR", mockTmpCacheDirectory.FullName }
                     });
                 // Unix uses TMPDIR as environment variable as opposed to TMP on windows
 
                 // Assert
+                if (StringComparer.Ordinal.Equals(cacheType, "all"))
```

Comment:
Conditions in a test should be thought through carefully (since multiple code paths are typically what need the testing!). You should test each locals directory individually wrt to the `cacheType` you parse out. For example, if `--clear global-packages` you would expected the `temp` local to _not_ be cleared, right?

nit: you can just `cacheType == "all"` here for brevity.